src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Unix.cs

```diff
@@ -2,108 +2,1037 @@
-        {
-            throw new PlatformNotSupportedException(SR.net_WebSockets_UnsupportedPlatform);
-        }
+                    // Add all of the required headers
+                    builder.Append("Host: ").Append(uri.Host).Append("\r\n");
+                    builder.Append("Connection: Upgrade\r\n");
+                    builder.Append("Upgrade: websocket\r\n");
+                    builder.Append("Sec-WebSocket-Version: 13\r\n");
+                    builder.Append("Sec-WebSocket-Key: ").Append(secKey).Append("\r\n");
 
-        public Task CloseOutputAsync(
-            WebSocketCloseStatus closeStatus,
-            string statusDescription,
-            CancellationToken cancellationToken)
-        {
-            throw new PlatformNotSupportedException(SR.net_WebSockets_UnsupportedPlatform);
-        }
+                    // Add all of the additionally requested headers
+                    foreach (string key in options.RequestHeaders.AllKeys)
+                    {
+                        builder.Append(key).Append(": ").Append(options.RequestHeaders[key]).Append("\r\n");
+                    }
 
-        public void Dispose()
-        {
-        }
+                    // Add the optional subprotocols header
+                    if (options.RequestedSubProtocols.Count > 0)
+                    {
+                        builder.Append(HttpKnownHeaderNames.SecWebSocketProtocol).Append(": ");
+                        builder.Append(options.RequestedSubProtocols[0]);
+                        for (int i = 1; i < options.RequestedSubProtocols.Count; i++)
+                        {
+                            builder.Append(", ").Append(options.RequestedSubProtocols[i]);
+                        }
+                        builder.Append("\r\n");
+                    }
 
-        public void Abort()
-        {
+                    // Add an optional cookies header
+                    if (options.Cookies != null)
+                    {
+                        string header = options.Cookies.GetCookieHeader(uri);
+                        if (!string.IsNullOrWhiteSpace(header))
+                        {
+                            builder.Append(HttpKnownHeaderNames.SetCookie).Append(": ").Append(header).Append("\r\n");
+                        }
+                    }
+
+                    // End the headers
+                    builder.Append("\r\n");
+
+                    // Return the bytes for the built up header
+                    return s_defaultHttpEncoding.GetBytes(builder.ToString());
+                }
+                finally
+                {
+                    // Make sure we clear the builder
+                    builder.Clear();
+                }
+            }
+
+            private static KeyValuePair<string, string> CreateSecKeyAndSecWebSocketAccept()
+            {
+                string secKey = Convert.ToBase64String(Guid.NewGuid().ToByteArray());
+                using (SHA1 sha = SHA1.Create())
```

Comment:
I expect that you considered saving a SHA1 instance into a (non-thread) static field and using Interlocked.Exchange to round-robin it (and Dispose of any parallel excesses).

On the off-chance that you didn't, consider it offered.
If you did, and didn't use it: I don't disagree. (Since I don't know the expected parallel factor of this code)

Or there's always instance method and make it an instance field, so it tears down sympathetically.